.gitlab-ci.yml

```diff
@@ -1,186 +1,4 @@
-  only:
-    variables:
-      - $SIMP_FULL_MATRIX == "yes"
-
-# Puppet Versions
-#-----------------------------------------------------------------------
-
-.pup_5: &pup_5
-  image: 'ruby:2.4'
-  variables:
-    PUPPET_VERSION: '~> 5.0'
-    BEAKER_PUPPET_COLLECTION: 'puppet5'
-    MATRIX_RUBY_VERSION: '2.4'
-
-.pup_5_5_10: &pup_5_5_10
-  image: 'ruby:2.4'
-  variables:
-    PUPPET_VERSION: '5.5.10'
-    BEAKER_PUPPET_COLLECTION: 'puppet5'
-    MATRIX_RUBY_VERSION: '2.4'
-
-.pup_6: &pup_6
-  image: 'ruby:2.5'
-  variables:
-    PUPPET_VERSION: '~> 6.0'
-    BEAKER_PUPPET_COLLECTION: 'puppet6'
-    MATRIX_RUBY_VERSION: '2.5'
-
-
-# Testing Environments
-#-----------------------------------------------------------------------
-
-.lint_tests: &lint_tests
-  stage: 'validation'
-  tags: ['docker']
-  <<: *setup_bundler_env
-  script:
-    - 'bundle exec rake syntax'
-    - 'bundle exec rake lint'
-    - 'bundle exec rake metadata_lint'
-
-.unit_tests: &unit_tests
-  stage: 'validation'
-  tags: ['docker']
-  <<: *setup_bundler_env
-  script:
-    - 'bundle exec rake spec'
-
-.acceptance_base: &acceptance_base
-  stage: 'acceptance'
-  tags: ['beaker']
-  <<: *setup_bundler_env
-
-.compliance_base: &compliance_base
-  stage: 'compliance'
-  tags: ['beaker']
-  <<: *setup_bundler_env
-
-
-# Pipeline / testing matrix
-#=======================================================================
-
-sanity_checks:
-  <<: *pup_5
-  <<: *setup_bundler_env
-  stage: 'sanity'
-  tags: ['docker']
-  script:
-    - 'if `hash apt-get`; then apt-get update; fi'
-    - 'if `hash apt-get`; then apt-get install -y rpm; fi'
-    - 'bundle exec rake check:dot_underscore'
-    - 'bundle exec rake check:test_file'
-    - 'bundle exec rake pkg:check_version'
-    - 'bundle exec rake pkg:compare_latest_tag'
-    - 'bundle exec rake pkg:create_tag_changelog'
-    - 'bundle exec puppet module build'
-
-# Linting
-#-----------------------------------------------------------------------
-
-pup5-lint:
-  <<: *pup_5
-  <<: *lint_tests
-
-pup6-lint:
-  <<: *pup_6
-  <<: *lint_tests
-
-# Unit Tests
-#-----------------------------------------------------------------------
-
-pup5-unit:
-  <<: *pup_5
-  <<: *unit_tests
-
-pup5.5.10-unit:
-  <<: *pup_5_5_10
-  <<: *unit_tests
-
-pup6-unit:
-  <<: *pup_6
-  <<: *unit_tests
-
-# Acceptance Tests
-#-----------------------------------------------------------------------
-
-pup5.5.10:
-  <<: *pup_5_5_10
-  <<: *acceptance_base
-  script:
-    - 'bundle exec rake beaker:suites'
-
-pup5.5.10-fips:
-  <<: *pup_5_5_10
-  <<: *acceptance_base
-  script:
-    - 'BEAKER_fips=yes bundle exec rake beaker:suites'
-
-pup5.5.10-oel:
-  <<: *pup_5_5_10
-  <<: *acceptance_base
-  script:
-    - 'bundle exec rake beaker:suites[default,oel]'
-
-pup6:
-  <<: *pup_6
-  <<: *acceptance_base
-  script:
-    - 'bundle exec rake beaker:suites'
+include:
+  - project: 'simp/remote-gitlab-ci'
+    file:    '/gitlab-templates/.basic.gitlab-ci.yml'
```

Comment:
(Following up on today's discussion:) Do not merge this yet.

This particular use of `include` and the current architecture of [simp/remote-gitlab-ci](https://github.com/simp/remote-gitlab-ci) will cause increasingly problematic RELENG-related issues over time.  This is a design issue, and it must be addressed _before_ introducing a centralized `include` strategy across all our repositories, or we may irretrievably lose the ability to run meaningful pipelines across different projects over time.

I'll follow up in more detail in issue [SIMP-5650](https://simp-project.atlassian.net/browse/SIMP-5650).